### PR TITLE
feat(validation): check template for available placeholders

### PR DESCRIPTION
**Pull request by winged**
This introduces the functionality that you can now pass a list of
available placeholders when creating a template. The template is then
checked to see whether it contains a strict subset of the given placeholders.

The syntax is a bit arbitrary, but allows specification of nested values,
and even lists:

    {
        "slug": ...,
        "available_placeholders": [
            "foo", "bar",
            "a_list[].property",
            "a_nested.object.with.property"
        ]
    }

### CODE REVIEW

**Comment by winged** [comment]
The `ListField` behaves in a strange way  when not using JSON. Works in the tests, but needs better documentation.

Also, the check is only implemented for Jinja2 templates (DOCX_TEMPLATE engine), not the mailmerge variant

**Comment by winged** [comment]
> Also, the check is only implemented for Jinja2 templates (DOCX_TEMPLATE engine), not the mailmerge variant

This has been implemented now as well.
Still needs some better docs though

**Comment by winged** [comment]
@czosel As discussed off-line, I've now completed the validation using sample data as well.

The documentation still needs a lot of love and more examples. I'll work on that next in a follow-up PR

File: README.md [hunk 1]
```diff
@@ -25,13 +25,52 @@ You can now access the api at [http://localhost:8000/api/v1/](http://localhost:8
...
 curl --form template=@docx-template.docx --form name="Test Template" --form engine=docx-template http://localhost:8000/api/v1/template/
 ```
 
-And merge template with:
+**Hint:**
+
+You can enable additional validation of the template against a list of available
+placeholders by filling the `available_placeholders` attribute:
+
+```json
```
Comment (winpat) [inline README.md]: So this means you can configure DMS to only allow the upload of templates with known placeholders?
If so, how do you configure it? Is there a curl command which could be documented?

File: README.md [hunk 2]
```diff
@@ -25,13 +25,52 @@ You can now access the api at [http://localhost:8000/api/v1/](http://localhost:8
...
+```
+
+The above would only allow templates that use the given list of placeholders,
+and reject templates that use an unknown placeholder name (such as "baz" in
+this case)
+
+Alternatively, you can also provide example data for a test render to
+the same effect, in JSON format:
+
+```json
```
Comment (winpat) [inline README.md]: The sample data is then the global default for all templates, correct?

File: README.md (hunk 1 above)
Comment (czosel) [reply README.md]: It's not config - it's part of the upload request - see #239

Comment (winpat) [reply README.md]: I see :+1:. I was a bit confused by the mix of form data and JSON in the same request.

File: README.md (hunk 2 above)
Comment (winpat) [reply README.md]: Got it.

**Pull request merged**